src/Build.UnitTests/BackEnd/BuildManager_Logging_Tests.cs

```diff
@@ -0,0 +1,214 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Build.BackEnd.Logging;
+using Microsoft.Build.Evaluation;
+using Microsoft.Build.Execution;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Shared;
+using Microsoft.Build.UnitTests;
+using Shouldly;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.NetCore.Extensions;
+using static Microsoft.Build.UnitTests.ObjectModelHelpers;
+
+#nullable disable
+
+namespace Microsoft.Build.Engine.UnitTests.BackEnd
+{
+    public class BuildManager_Logging_Tests : IDisposable
+    {
+        private string mainProject = @"
```

Comment:
nit: this and the `childProjectWithCustomBuildEvent` should be `const` and name start with `_` (to follow the suit of other fields in this file)

Reply:
unfortunately, I can't make them constants
https://learn.microsoft.com/en-us/dotnet/csharp/misc/cs0133?f1url=%3FappId%3Droslyn%26k%3Dk(CS0133)

Reply:
Oh right. They could be `readonly`.
But it doesn't matter in test code